### ISSUE TITLE
Get-MgUser -Filter "mail eq 'email@email.com'" | select -ExpandProperty Drive does not return drive info

### ISSUE THREAD

**Issue by rogers14k** [microsoftgraph/msgraph-sdk-powershell#2105]
Get-MgUser -Filter "mail eq 'email@email.com'" | select -ExpandProperty Drive does not return drive info but there is an active onedrive for the searched users. There has been for years and there are files/folders shared from them.
![Screenshot 2023-06-22 at 3 06 33 PM](https://github.com/microsoftgraph/msgraph-sdk-powershell/assets/64924757/0eb1bd53-686f-4a65-b359-2a63af5d9877)

**Comment by SeniorConsulting** [comment]
Hi Rogers,

It's not entirely clear until you do some digging to figure out what's required here. There are multiple things at play here (as I understand it):

1. You have to use a service principal (application) to query the data. You cannot use delegated permissions, so when connecting to Graph, you'll need to either use a Managed Identity or other app registration. This is loosely referenced [here](https://learn.microsoft.com/en-us/graph/permissions-reference#remarks-6) and [here](https://learn.microsoft.com/en-us/graph/api/drive-get?view=graph-rest-1.0&tabs=http#get-a-users-onedrive)
2. In addition to User.Read permission, you'll also need (at minimum), Files.Read.All
3. You won't be able to use the Get-MgUser in the way you were expecting, as it's a little bit different from that, but fear not, it should still be a single cmdlet: Get-MgUserDefaultDrive 

So, in summary: Create use a service principal, grant it Files.Read.All, and use Get-MgUserDefaultDrive

Hopefully that helps.

**Comment by rogers14k** [comment]
> Hi Rogers,
> 
> It's not entirely clear until you do some digging to figure out what's required here. There are multiple things at play here (as I understand it):
> 
>     1. You have to use a service principal (application) to query the data. You cannot use delegated permissions, so when connecting to Graph, you'll need to either use a Managed Identity or other app registration. This is loosely referenced [here](https://learn.microsoft.com/en-us/graph/permissions-reference#remarks-6) and [here](https://learn.microsoft.com/en-us/graph/api/drive-get?view=graph-rest-1.0&tabs=http#get-a-users-onedrive)
> 
>     2. In addition to User.Read permission, you'll also need (at minimum), Files.Read.All
> 
>     3. You won't be able to use the Get-MgUser in the way you were expecting, as it's a little bit different from that, but fear not, it should still be a single cmdlet: Get-MgUserDefaultDrive
> 
> 
> So, in summary: Create use a service principal, grant it Files.Read.All, and use Get-MgUserDefaultDrive
> 
> Hopefully that helps.

Thank you. I missed that!
